### DISCUSSION TITLE
[!] 'use_app_entitlements' value must be either `true` or `false`! Found String instead

### DISCUSSION THREAD

**Discussion by ArturRuZ** [fastlane/fastlane#19120] [Q&A] [answered]
After update fastlane, I can't resign ipa with use_app_entitlements key. I get error:

You passed invalid parameters to 'resign'.
Check out the error below and available options by running `fastlane action resign`

[!] 'use_app_entitlements' value must be either `true` or `false`! Found String instead. 

I use this command: 
fastlane run resign ipa:"PATH/origin.ipa" signing_identity:"MY EDENTITY" provisioning_profile:"PATH/distribution.mobileprovision" bundle_id:my.bundle use_app_entitlements:"PATH/default.ipa"

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.15.7                                     |
| Ruby                        | 2.7.2                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.24.3 (Apple Git-128)          |
| Installation Source         | ~/.rvm/gems/ruby-2.7.2/bin/fastlane         |
| Host                        | Mac OS X 10.15.7 (19H1030)                  |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.7.2/lib                |
| OpenSSL Version             | OpenSSL 1.1.1k  25 Mar 2021                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 12.4                                        |
| Swift Version               | 5.3.2                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | ru_RU.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.187.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                             | Version      |
| ------------------------------- | ------------ |
| did_you_mean                    | 1.4.0        |
| executable-hooks                | 1.6.1        |
| bundler-unload                  | 1.0.2        |
| rubygems-bundler                | 1.4.5        |
| bundler                         | 2.2.16       |
| atomos                          | 0.1.3        |
| CFPropertyList                  | 3.0.3        |
| claide                          | 1.0.3        |
| colored2                        | 3.1.2        |
| nanaimo                         | 0.3.0        |
| xcodeproj                       | 1.19.0       |
| rouge                           | 2.0.7        |
| xcpretty                        | 0.3.0        |
| terminal-notifier               | 2.0.0        |
| unicode-display_width           | 1.7.0        |
| terminal-table                  | 1.8.0        |
| plist                           | 3.6.0        |
| public_suffix                   | 4.0.6        |
| addressable                     | 2.7.0        |
| multipart-post                  | 2.0.0        |
| word_wrap                       | 1.0.0        |
| tty-screen                      | 0.8.1        |
| tty-cursor                      | 0.7.1        |
| tty-spinner                     | 0.9.3        |
| artifactory                     | 3.0.15       |
| babosa                          | 1.0.4        |
| colored                         | 1.2          |
| highline                        | 2.0.3        |
| commander                       | 4.6.0        |
| excon                           | 0.79.0       |
| faraday-net_http                | 1.0.1        |
| ruby2_keywords                  | 0.0.4        |
| faraday                         | 1.3.0        |
| unf_ext                         | 0.0.7.7      |
| unf                             | 0.1.4        |
| domain_name                     | 0.5.20190701 |
| http-cookie                     | 1.0.3        |
| faraday-cookie_jar              | 0.0.7        |
| faraday_middleware              | 1.0.0        |
| fastimage                       | 2.2.3        |
| gh_inspector                    | 1.1.3        |
| json                            | 2.3.0        |
| mini_magick                     | 4.11.0       |
| naturally                       | 2.2.1        |
| rubyzip                         | 2.3.0        |
| security                        | 0.1.3        |
| xcpretty-travis-formatter       | 1.0.1        |
| dotenv                          | 2.7.6        |
| simctl                          | 1.6.8        |
| jwt                             | 2.2.2        |
| google-cloud-env                | 1.5.0        |
| google-cloud-errors             | 1.1.0        |
| google-cloud-core               | 1.6.0        |
| google-apis-iamcredentials_v1   | 0.3.0        |
| google-apis-storage_v1          | 0.3.0        |
| rake                            | 13.0.1       |
| digest-crc                      | 0.6.3        |
| mini_mime                       | 1.1.0        |
| google-cloud-storage            | 1.31.0       |
| emoji_regex                     | 3.2.2        |
| jmespath                        | 1.4.0        |
| aws-partitions                  | 1.443.0      |
| aws-eventstream                 | 1.1.1        |
| aws-sigv4                       | 1.2.3        |
| aws-sdk-core                    | 3.113.1      |
| aws-sdk-kms                     | 1.43.0       |
| aws-sdk-s3                      | 1.93.0       |
| uri                             | 0.10.0       |
| webrick                         | 1.6.0        |
| rexml                           | 3.2.3        |
| httpclient                      | 2.8.3        |
| multi_json                      | 1.15.0       |
| signet                          | 0.15.0       |
| os                              | 1.1.1        |
| memoist                         | 0.16.2       |
| googleauth                      | 0.16.2       |
| retriable                       | 3.1.2        |
| trailblazer-option              | 0.1.0        |
| declarative                     | 0.0.20       |
| uber                            | 0.1.0        |
| representable                   | 3.1.0        |
| google-apis-core                | 0.4.1        |
| google-apis-playcustomapp_v1    | 0.5.0        |
| google-apis-androidpublisher_v3 | 0.8.0        |
| forwardable                     | 1.3.1        |
| logger                          | 1.4.2        |
| cgi                             | 0.1.0        |
| date                            | 3.0.0        |
| timeout                         | 0.1.0        |
| stringio                        | 0.1.0        |
| zlib                            | 1.1.0        |
| ipaddr                          | 1.2.2        |
| openssl                         | 2.1.2        |
| ostruct                         | 0.2.0        |
| strscan                         | 1.0.3        |
| io-console                      | 0.5.6        |
| delegate                        | 0.1.0        |
| fileutils                       | 1.4.1        |
| singleton                       | 0.1.0        |
| open3                           | 0.1.0        |
| yaml                            | 0.1.0        |
| psych                           | 3.1.0        |
| mutex_m                         | 0.1.0        |
</details>

*generated on:* **2021-07-21**
</details>

it's a bug?

**Comment by crazymanish** [comment, marked as answer]
Hi @ArturRuZ 

What happens if you pass `true` for use_app_entitlements param? 🤔   i.e
```ruby
fastlane run resign ipa:"PATH/origin.ipa" signing_identity:"MY EDENTITY" provisioning_profile:"PATH/distribution.mobileprovision" bundle_id:my.bundle use_app_entitlements:true

**Reply by crazymanish** [reply]
btw,
In this PR https://github.com/fastlane/fastlane/pull/18962 I changed this to accept `true or false` because that is what is expected. The earlier passing of a string for this param might be treated as `true` but unfortunately, now we need to pass the Bool value explicitly! 
Please see for more details, it should be just a Bool value flag
https://github.com/fastlane/fastlane/blob/master/sigh/lib/sigh/resign.rb#L54


Please note: If you want to pass the **explicit entitlements path**, please use this option https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/resign.rb#L61 

I hope this will help! 🙏 
Please let me know in case of any issue.

**Reply by ArturRuZ** [reply]
@crazymanish  I was wrong. Now it works with true. But I don't understand how to  pass entitlements path?

**Reply by crazymanish** [reply]
Please note: If you want to pass the **explicit entitlements path**, please use below option https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/resign.rb#L61

```ruby
fastlane run resign 
ipa:"PATH/origin.ipa" 
signing_identity:"MY EDENTITY" 
provisioning_profile:"PATH/distribution.mobileprovision" 
bundle_id:my.bundle 
entitlements: "PATH/MyApp.entitlements" # this is how you can pass 😇

**Reply by ArturRuZ** [reply]
and in this case I shouldn't use use_app_entitlements:true?

**Reply by crazymanish** [reply]
Yes, between these two options, **You should use always one of them**

```ruby
# either use 
use_app_entitlements:true
# or 
entitlements: "PATH/MyApp.entitlements"
```

Also you try to use both params same time, Fastlane will raise error because that is invalid configuration

**Reply by ArturRuZ** [reply]
oh, I understand. In new version  **use_app_entitlements:true** gets entitlements from resigining app. **entitlements** - uses another entitlements for resigning app?
